File: Stack/cpp/myStack.cpp
#include "myStack.h"

myStack::myStack(int length) {
    this->length = length;
    items = new int[length];
    count = 0;
}

void myStack::push(int item) {
    if (count == length)
        return;
    items[count++] = item;
}

int myStack::pop() {
    if (0 == count)
        exit(EXIT_FAILURE);
    return items[--count];
}

int myStack::peek() {
    if (0 == count)
        exit(EXIT_FAILURE);
    return items[count - 1];
}

bool myStack::isEmpty() {
    return 0 == count;
}

File: Stack/java/src/myStack.java
import java.util.Arrays;

public class myStack {
    private int[] items = new int[5];
    private int count;

    public void push(int item) {
        if (count == items.length)
            throw new StackOverflowError();

        items[count++] = item;
    }

    public int pop() {
        if (0 == count)
            throw new IllegalStateException();

        return items[--count];
    }

    public int peek() {
        if (0 == count)
            throw new IllegalStateException();

        return items[count - 1];
    }

    public boolean isEmpty() {
        return 0 == count;
    }

    @Override
    public String toString() {
        var content = Arrays.copyOfRange(items, 0, count);
        return Arrays.toString(content);
    }


}
